sklearn/preprocessing/label.py

```diff
@@ -472,9 +473,21 @@ def label_binarize(y, classes, neg_label=0, pos_label=1, sparse_output=False):
 
     if y_type == "binary":
         if len(classes) == 1:
-            Y = np.zeros((len(y), 1), dtype=np.int)
-            Y += neg_label
-            return Y
+            if sparse_output:
+                if isinstance(classes[0], integer_types) and classes[0] != 0:
+                    classes = np.sort(np.append(classes, 0))
```

Comment:
Rather than appending 0 at the end and sorting, I would do "classes = np.r_[0, classes]" which avoid doing 2 array operations.